### PR TITLE
Added interation tests to validate example run successfully

### PR DESCRIPTION
**Pull request by hemantdwave**
Please review added integration tests to validate example

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import unittest
+import time
+import os
+import sys
+
+project_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+class IntegrationTests(unittest.TestCase):
+
+    def test_sudoku(self):
```
Comment (randomir) [inline tests/test_integration.py:25]: sudoku

File: tests/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        example_file = os.path.join(project_dir, 'graph_coloring.py')
+        output = subprocess.check_output([sys.executable, example_file])
+        output = str(output).upper()
+        if os.getenv('DEBUG_OUTPUT'):
+            print("Example output \n" + output)
+
+        with self.subTest(msg="Verify if output contains valid solution energy:  5.0' \n"):
+            self.assertIn("Solution energy:  5.0".upper(), output)
+        with self.subTest(msg="Verify if solution validity:  True \n"):
+            self.assertIn("Solution validity:  True".upper(), output)
```
Comment (randomir) [inline tests/test_integration.py:35]: Is this the only acceptable solution? /cc @joelgdwave

File: tests/test_integration.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import time
+import os
+import sys
+
+project_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+class IntegrationTests(unittest.TestCase):
+
+    def test_sudoku(self):
+        example_file = os.path.join(project_dir, 'graph_coloring.py')
```
Comment (randomir) [inline tests/test_integration.py:26]: This is not example file, but the python program file.

**Comment by hemantdwave** [comment]
Updated as per review comments

**Review by randomir — COMMENTED** [review]
Looks good. I'm just still wondering should we hard-code `5.0` as the only valid solution energy. Thoughts, @joelgdwave?

**Comment by joelgdwave** [comment]
Yes, 5.0 is the only valid solution energy for this problem. It's a very basic problem. I've never seen any other energies. I have not studied it over a long period of time, though.

**Comment by randomir** [comment]
Thanks, @joelgdwave!

**Pull request merged**